client/src/pages/Duel.js

```diff
@@ -0,0 +1,200 @@
+    axios
+      .get(`/api/v1/duels/${duelId}`)
+      .then((res) => {
+        if (res.data.inProgress === 0) {
+          document.location.replace("/homepage");
+          return;
+        }
+        this.setState({
+          duelData: res.data,
+        });
+
+        // Can't access to a round already played
+        const firstQuestion = res.data.rounds[res.data.currentRound - 1][0];
+        console.log(firstQuestion);
+        if (firstQuestion.userAnswer !== undefined) {
+          document.location.replace("/homepage");
+        }
+      })
+      .catch((err) => console.error(err));
+
+  }
+
+  /**
+   * Get data about current round
+   */
+  getCurrentRound() {
+    const { duelData } = this.state;
+    return duelData.rounds[duelData.currentRound - 1];
+  }
+
+  /**
+   * Get data about current question
+   */
+  getCurrentQuestion() {
+    const { currentQuestionNum } = this.state;
+    return this.getCurrentRound()[currentQuestionNum - 1];
+  }
+
+  /**
+   * Update timer
+   * @param {int} value Timer value
+   */
+  updateTimer = (value) => {
+    let { inProgress, userAnswers } = this.state;
+
+    if (!inProgress) return false;
+    if (value === 0) {
+      userAnswers.push(-1);
+      inProgress = false;
+    }
+
+    this.setState({
+      inProgress: inProgress,
+      timer: value,
+      userAnswers: userAnswers,
+    });
+  };
+
+  /**
+   * Handle click on an answer
+   * @param {string} value Answer value
+   */
+  handleAnswerClick = (value) => {
+    if (!this.state.inProgress) return;
+
+    const { userAnswers } = this.state;
+    const currentQuestion = this.getCurrentQuestion();
+    userAnswers.push(currentQuestion.answers.indexOf(value));
+
+    this.setState({
+      inProgress: false,
+      lastClicked: value,
+      userAnswers: userAnswers,
+    });
+  };
+
+  /**
+   * Get next question or end the duel if it is the last question.
+   */
+  nextQuestion = () => {
+    const { currentQuestionNum } = this.state;
+
+    if (currentQuestionNum === this.getCurrentRound().length) {
+      this.validateDuel();
+      return;
+    }
+
+    this.setState({
+      inProgress: true,
+      currentQuestionNum: currentQuestionNum + 1,
+      timer: Duel.TIMER_DURATION,
+    });
+  };
+
+  /**
+   * Compute the class of answer indicators at the top of the page
+   * @param {int} index Index of the answer
+   * @param {object} question Question data
+   */
+  computeClassDuelBar(index, question) {
+    const { userAnswers } = this.state;
+    if (index >= userAnswers.length) {
+      return "";
+    }
+    return question.goodAnswer === userAnswers[index] ? "good" : "bad";
+  }
+
+  /**
+   *Sends user answers to the server
+   */
+  validateDuel = () => {
+    const { duelData, userAnswers } = this.state;
+
+    axios
+      .post(`/api/v1/duels/${duelData.id}/${duelData.currentRound}`, {
+        answers: userAnswers,
+      })
+      .then(() => document.location.replace(`/duel/${duelData.currentRound}`))
```

Comment:
```suggestion

    const duelId = this.props.match.params.id;
    axios
      .post(`/api/v1/duels/${duelData.id}/${duelData.currentRound}`, {
        answers: userAnswers,
      })
      .then(() => document.location.replace(`/duel/${duelId}`))
```

Reply:
Why? Both work

Reply:
No, `currentRound` is the number of the round in the current duel.
It's different from the duelID.

Reply:
Oh yes!

Reply:
The right variable was duelData.id instead of dualData.currentRound but I emmêled my pinceaux

Reply:
No problem, it's sometimes hard to paint while listening to someone